src/test/analyzer_test.ts

```diff
@@ -84,6 +85,32 @@ suite('Analyzer', () => {
           ['MyNamespace.SubBehavior', 'MyNamespace.SimpleBehavior']);
     });
 
+    // TODO(fks) 09-28-2016: Fix off-by-one error in warning sourceRange line
+    // numbers.
+    test('throws when cant find behavior', async() => {
```

Comment:
I tend towards more integration tests and fewer unit tests myself, so I think this is great. 

It seems like otherwise you'd need to either set up parsing, scanning, resolving, etc yourself, ending up with your own mini-analyzer just for testing, or you'd have to  hand-constructing complex middle-stage objects. Both options seem too unreliable and divergent from the code as it is used.